sda-download/internal/storage/storage.go

```diff
@@ -315,3 +351,439 @@ func transportConfigS3(config S3Conf) http.RoundTripper {
 
 	return trConfig
 }
+
+func (sb *s3SeekableBackend) NewFileReader(filePath string) (io.ReadCloser, error) {
+	s := sb.s3Backend
+	objectSize, err := s.GetFileSize(filePath)
+
+	if err != nil {
+		return nil, err
+	}
+
+	reader := &s3SeekableReader{
+		*sb,
+		0,
+		make([]s3CacheBlock, 0, 32),
+		filePath,
+		objectSize,
+		sync.Mutex{},
+		make([]int64, 0, 32),
+		false,
+		nil,
+	}
+
+	return reader, nil
+}
```

Comment:
Wouldn't the implementation of this as a `SeekableFileReader` in the normal backend be the same, so we can do both types of file reads depending on the incoming query.

Reply:
True. I made this in a way to keep the `s3` driver unchanged for now, but in an ideal future scenario, I would like to see a single driver (supporting seeking), possibly offering some more controls to adjust how it works.